pkg/kubelet/kubelet_pods.go

```diff
@@ -206,7 +206,7 @@ func makeHostsMount(podDir, podIP, hostName, hostDomainName string) (*kubecontai
 // ensureHostsFile ensures that the given host file has an up-to-date ip, host
 // name, and domain name.
 func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string) error {
-	if _, err := os.Stat(fileName); os.IsExist(err) {
+	if _, err := os.Stat(fileName); err == nil || os.IsExist(err) {
```

Comment:
Yes, you are right. [Here](https://gist.github.com/mastef/05f46d3ab2f5ed6a6787#file-isexist_vs_isnotexist-go-L33-L64) is a detailed analysis.